### PR TITLE
add isInsufficientMaterial method

### PR DESCRIPTION
**Pull request by vladmandic**
adds method to detect insufficient material to end game in checkmate under both FIDE and USCF rules.  
further notes in previous version of the pr #34

### CODE REVIEW

File: src/position.ts [hunk 1]
```diff
@@ -663,6 +663,20 @@ export class Position {
 	}
 
 
+	/**
+	 * Whether both players have insufficient material so the game cannot end in checkmate. If the position is not legal (see {@link Position.isLegal}),
```
Comment (yo35) [inline src/position.ts:667]: The check on `isLegal(..)` is missing in the method implementation.

**Review by yo35 — CHANGES_REQUESTED** [review]

File: src/position.ts [hunk 2]
```diff
@@ -663,6 +663,20 @@ export class Position {
...
+	 * the returned value is always `false`.
+   *
+   * If `forcedMate` is not set or set to `false`, method uses **FIDE** rules where position is evaluated to return no possible checkmates
+   * If `forcedMate` is set to `true`, method uses **USCF** rules where position is evaluated to return no possible *forced* checkmates
+	 *
+	 * For antichess and horde chess, this method always returns `false`
+	 */
+	isInsufficientMaterial(forcedMate?: boolean): boolean {
```
[the remark this answers is not in the document]
Comment (vladmandic) [reply src/position.ts]: done.

File: src/position.ts (hunk 1 above)
Comment (vladmandic) [reply src/position.ts:667]: done.

File: src/private_position/move_generation.ts [hunk 3]
```diff
@@ -143,6 +143,36 @@ export function isStalemate(position: PositionImpl) {
...
 
 
+/**
+ * Whether the given position is level and there is insufficient material on board.
+ */
+export function isInsufficientMaterial(position: PositionImpl, forcedMate?: boolean) {
+	if (position.variant === GameVariantImpl.ANTICHESS) {
+		return false;
+	}
+	else if (position.variant === GameVariantImpl.HORDE && position.turn === ColorImpl.WHITE) {
```
[the remark this answers is not in the document]
Comment (vladmandic) [reply src/private_position/move_generation.ts]: done. i'm not familar with horde.

File: src/private_position/move_generation.ts [hunk 4]
```diff
@@ -143,6 +143,36 @@ export function isStalemate(position: PositionImpl) {
...
+		return false;
+	}
+	else {
+		const FEN_PIECE_SYMBOL = [ ...'KkQqRrBbNnPp' ];
+		const DEF_INSUFFICIENT_FIDE = ['kk', 'kkn', 'kbk', 'knkn']; // no possible mate
```
[the remark this answers is not in the document]
Comment (vladmandic) [reply src/private_position/move_generation.ts]: done. my mistake.

**Review by yo35 — APPROVED** [review]

**Comment by yo35** [comment]
For information, I've added some unit tests (see in particular in file `/test/resources/positions.csv`, rows `Low material *`), and I had to modify your implementation of `isDead()` to make them work, especially in situations with several same-colored bishops (e.g. king + 2 same-colored bishops vs king is a dead position).

Public API is unchanged compared to the PR.

**Pull request merged**